lib/components/ImageButton.tsx

```diff
@@ -0,0 +1,226 @@
+/**
+ * @file
+ * @copyright 2024 Aylong (https://github.com/AyIong)
+ * @license MIT
+ */
+
+import { Placement } from '@popperjs/core';
+import { ReactNode } from 'react';
+
+import { BooleanLike, classes } from '../common/react';
+import styles from '../styles/components/ImageButton.module.scss';
+import { BoxProps, computeBoxProps } from './Box';
+import { DmIcon } from './DmIcon';
+import { Icon } from './Icon';
+import { Image } from './Image';
+import { Stack } from './Stack';
+import { Tooltip } from './Tooltip';
+
+type Props = Partial<{
+  /** Asset cache. Example: `asset={['assetname32x32', thing.key]}` */
+  asset: string[];
+  /** Classic way to put images. Example: `base64={thing.image}` */
+  base64: string;
+  /** 
+   * Special container for buttons.
+   * You can put any other component here.
+   * Has some special stylings!
+   * Example: `buttons={<Button>Send</Button>}` 
+   */
+  buttons: ReactNode;
+  /** Enables alternate buttons container. */
+  buttonsAlt: boolean;
+  /** Content under image. Or on the right if fluid. */
+  children: ReactNode;
+  /** Applies a CSS class to the element. */
+  className: string;
+  /** Color of the button. See [Button](#button) but without `transparent`. */
+  color: string;
+  /** Makes button disabled and dark red if true. Also disables onClick. */
+  disabled: BooleanLike;
+  /** Optional. Adds a "stub" when loading DmIcon. */
+  dmFallback: ReactNode;
+  /** Parameter `icon` of component `DmIcon`. */
+  dmIcon: string | null;
+  /** Parameter `icon_state` of component `DmIcon`. */
+  dmIconState: string | null;
+  /** 
+   * Changes the layout of the button, making it fill the entire horizontally available space.
+   * Allows the use of `title` 
+   */
+  fluid: boolean;
+  /** Parameter responsible for the size of the image, component and standard "stubs". */
+  imageSize: number;
```

Comment:
why enforce this has to be in `px` instead of letting them pass a string size

Reply:
Image sizes standardized to be in px, read like sprite size
In my opinion it is much more convenient to write just `imageSize={64}` instead of `imageSize=‘64px’`.